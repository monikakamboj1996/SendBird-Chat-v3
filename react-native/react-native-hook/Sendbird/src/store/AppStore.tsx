import AsyncStorage from '@react-native-async-storage/async-storage';
import { createStore, applyMiddleware } from 'redux';

import { persistStore, persistReducer } from 'redux-persist';
import thunk from 'redux-thunk';
import RootReducer from '../reducer/RootReducer';

// Middleware: Redux Persist Config
const persistConfig = {
  // Root
  key: 'root',
  // Storage Method (React Native)
  storage: AsyncStorage,
};

// Middleware: Redux Persist Persisted Reducer
const persistedReducer = persistReducer(persistConfig, RootReducer);

// Redux: Store
const store = createStore(persistedReducer, applyMiddleware(thunk));

// Middleware: Redux Persist Persister
let persistor = persistStore(store);
// Exports
export { store, persistor };
