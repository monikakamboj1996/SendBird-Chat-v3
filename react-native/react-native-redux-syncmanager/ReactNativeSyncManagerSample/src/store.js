import {createStore, compose, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducers from './reducer';

export const store = createStore(reducers, {}, compose(applyMiddleware(thunk)));
