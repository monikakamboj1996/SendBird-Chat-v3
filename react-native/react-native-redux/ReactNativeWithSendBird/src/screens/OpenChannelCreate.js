import React, { Component } from 'react';
import { View, Text, TextInput, ScrollView } from 'react-native';
import { connect } from 'react-redux';
import {
  createOpenChannel,
  initOpenChannelCreate,
  addOpenChannelItem,
  openChannelProgress,
  onOpenChannelPress
} from '../actions';
import { Button, Input, Spinner, FormValidationMessage } from '../components';
import SendBird from 'sendbird';

class OpenChannelCreate extends Component {
  static navigationOptions = ({ navigation }) => {
    const { params } = navigation.state;
    return {
      title: 'Open Channel Create',
      headerLeft: (
        <Button
          containerViewStyle={{ marginLeft: 0, marginRight: 0 }}
          buttonStyle={{ paddingLeft: 14 }}
          icon={{
            name: 'chevron-left',
            type: 'font-awesome',
            color: '#7d62d9',
            size: 18
          }}
          backgroundColor="transparent"
          onPress={() => navigation.goBack()}
        />
      ),
      headerRight: (
        <Button
          containerViewStyle={{ marginLeft: 0, marginRight: 0 }}
          buttonStyle={{ paddingRight: 14 }}
          color={'#7d62d9'}
          title="create"
          backgroundColor="transparent"
          onPress={() => {
            params.handleHeaderRight();
          }}
        />
      )
    };
  };

  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      channelName: ''
    };
  }

  componentDidMount() {
    this.props.navigation.setParams({
      handleHeaderRight: this._onCreateButtonPress
    });
    this.props.initOpenChannelCreate();
  }

  componentDidUpdate(prevProps, prevState, snapshot) {
    const { channel, error } = this.props;

    if (channel && this.props.channel !== prevProps.channel) {
      this.props.openChannelProgress(true);
      this.setState({ isLoading: false }, () => {
        this.props.addOpenChannelItem(channel);
        this.props.navigation.goBack();
        this.props.onOpenChannelPress(channel.url);
      });
    }

    if (error) {
      this.setState({ isLoading: false });
    }
  }

  _onOpenChannelNameChanged = channelName => {
    this.setState({ channelName });
  };

  _onCreateButtonPress = () => {
    const { channelName } = this.state;
    this.setState({ isLoading: true }, () => {
      this.props.createOpenChannel(channelName);
    });
  };

  render() {
    return (
      <View style={{ backgroundColor: '#fff', flex: 1 }}>
        <Spinner visible={this.state.isLoading} />
        <View style={{ margin: 14, marginTop: 100 }}>
          <ScrollView style={styles.inputViewStyle}>
            <TextInput
              textFocusColor={'#7d62d9'}
              textBlurColor={'#000'}
              highlightColor={'#7d62d9'}
              borderColor={'#d9d9d9'}
              style={styles.inputStyle}
              keyboardType={'default'}
              autoCapitalize="none"
              duration={100}
              autoCorrect={false}
              placeholder="Channel Name"
              value={this.state.channelName}
              maxLength={12}
              onChangeText={this._onOpenChannelNameChanged}
            />
          </ScrollView>
        </View>
        <FormValidationMessage labelStyle={{ marginLeft: 14 }}>{this.props.error}</FormValidationMessage>
      </View>
    );
  }
}

function mapStateToProps({ openChannelCreate }) {
  const { error, channel } = openChannelCreate;
  return { error, channel };
}

export default connect(
  mapStateToProps,
  {
    createOpenChannel,
    initOpenChannelCreate,
    addOpenChannelItem,
    openChannelProgress,
    onOpenChannelPress
  }
)(OpenChannelCreate);

const styles = {
  inputViewStyle: {
    borderWidth: 1,
    borderColor: '#ddd',
    borderRadius: 5,
    paddingLeft: 8,
    paddingRight: 8,
    marginLeft: 28,
    marginRight: 28,
    marginTop: 8
  },
  inputStyle: {
    fontSize: 13,
    backgroundColor: '#fff'
  }
};
