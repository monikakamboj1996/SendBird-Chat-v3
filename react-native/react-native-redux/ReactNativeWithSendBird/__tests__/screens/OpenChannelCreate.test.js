import 'react-native';
import React from 'react';
import OpenChannelCreate from '../../src/screens/OpenChannelCreate';
import store from '../../src/store';
import renderer from 'react-test-renderer';

const navigation = {
  setParams: jest.fn(),
  navigate: jest.fn(),
  state: {
    params: {
      channelUrl: 'channelUrl'
    }
  },
  dispatch: jest.fn()
};

describe('screen/OpenChannelCreate', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<OpenChannelCreate store={store} navigation={navigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO: add test cases
});
