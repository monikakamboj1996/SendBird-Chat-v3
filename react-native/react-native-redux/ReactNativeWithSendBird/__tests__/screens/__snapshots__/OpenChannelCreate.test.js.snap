// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen/OpenChannelCreate renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "margin": 14,
        "marginTop": 100,
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "borderColor": "#ddd",
          "borderRadius": 5,
          "borderWidth": 1,
          "marginLeft": 28,
          "marginRight": 28,
          "marginTop": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          borderColor="#d9d9d9"
          duration={100}
          highlightColor="#7d62d9"
          keyboardType="default"
          maxLength={12}
          onChangeText={[Function]}
          placeholder="Channel Name"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#fff",
              "fontSize": 13,
            }
          }
          textBlurColor="#000"
          textFocusColor="#7d62d9"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#ff190c",
              "fontSize": 15,
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 5,
            },
            Object {
              "marginLeft": 14,
            },
            undefined,
          ],
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
